client/my-sites/checkout/checkout/web-payment-box.jsx

```diff
@@ -0,0 +1,442 @@
+import { setPayment } from 'lib/upgrades/actions';
+import {
+	BEFORE_SUBMIT,
+	INPUT_VALIDATION,
+	RECEIVED_PAYMENT_KEY_RESPONSE,
+	RECEIVED_WPCOM_RESPONSE,
+	SUBMITTING_PAYMENT_KEY_REQUEST,
+	SUBMITTING_WPCOM_REQUEST,
+} from 'lib/store-transactions/step-types';
+
+/**
+ * Web Payment method identifier.
+ */
+export const WEB_PAYMENT_BASIC_CARD_METHOD = 'basic-card';
+export const WEB_PAYMENT_APPLE_PAY_METHOD = 'https://apple.com/apple-pay';
+
+/**
+ *
+ */
+const SUPPORTED_NETWORKS = [ 'visa', 'mastercard', 'amex' ];
+const APPLE_PAY_MERCHANT_IDENTIFIER = 'merchant.com.wordpress.test';
+const PAYMENT_REQUEST_OPTIONS = {
+	requestPayerName: true,
+	requestPayerPhone: false,
+	requestPayerEmail: false,
+	requestShipping: true,
+};
+
+/**
+ * Returns an available Web Payment method.
+ *
+ * Web Payments (`PaymentRequest` API) are available only if the
+ * document is served through HTTPS.
+ *
+ * The configuration features `my-sites/checkout/web-payment/*` must
+ * also be enabled.
+ *
+ * @returns {string|null}  One of the `WEB_PAYMENT_*_METHOD` or `null`
+ *                         if none can be detected.
+ */
+export function detectWebPaymentMethod() {
+	if ( ! config.isEnabled( 'my-sites/checkout/web-payment' ) ) {
+		return null;
+	}
+
+	if (
+		config.isEnabled( 'my-sites/checkout/web-payment/apple-pay' ) &&
+		window.ApplePaySession &&
+		window.ApplePaySession.canMakePayments()
+	) {
+		return WEB_PAYMENT_APPLE_PAY_METHOD;
+	}
+
+	if ( config.isEnabled( 'my-sites/checkout/web-payment/basic-card' ) && window.PaymentRequest ) {
+		return WEB_PAYMENT_BASIC_CARD_METHOD;
+	}
+
+	return null;
+}
+
+/**
+ * Returns a user-friendly Web Payment method name.
+ *
+ * @param {string|null} webPaymentMethod  The payment method identifier
+ *                                        (expecting one of the
+ *                                        `WEB_PAYMENT_*_METHOD`
+ *                                        constant).
+ * @returns {string|null}                 A user-friendly payment name
+ *                                        or the given payment method
+ *                                        if none matches.
+ */
+export function getWebPaymentMethodName( webPaymentMethod ) {
+	switch ( webPaymentMethod ) {
+		case WEB_PAYMENT_BASIC_CARD_METHOD:
+			return 'Browser wallet';
+
+		case WEB_PAYMENT_APPLE_PAY_METHOD:
+			return ' Apple Pay';
+
+		default:
+			return webPaymentMethod;
+	}
+}
+
+/**
+ * The `<WebPaymentBox />` component.
+ */
+export class WebPaymentBox extends React.Component {
+	static propTypes = {
+		cart: PropTypes.object.isRequired,
+		transaction: PropTypes.object.isRequired,
+		transactionStep: PropTypes.object.isRequired,
+		onSubmit: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	buttonState = () => {
+		const { transactionStep, translate } = this.props;
+
+		const defaultState = () => {
+			return {
+				disabled: false,
+				text: translate( 'Select a payment card', {
```

Comment:
:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?